### PR TITLE
Return avg contributor engagement in organization profile

### PR DESCRIPTION
**Pull request by yeganathan18**
# Changes proposed ✍️

### What
copilot:summary
​
copilot:poem

### How
copilot:walkthrough

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [x] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: backend/src/database/migrations/U1719390926__orgAvgContributorEngagement.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE "organizationSegmentsAgg" DROP COLUMN "avgContributorEngagement";
```
Comment (sausage-todd) [inline backend/src/database/migrations/U1719390926__orgAvgContributorEngagement.sql:1]: We don't use rollbacks, so no need in these

File: backend/src/database/migrations/V1719390926__orgAvgContributorEngagement.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+ALTER TABLE "organizationSegmentsAgg" ADD COLUMN "avgContributorEngagement" INTEGER DEFAULT '-1'::INTEGER;
```
Comment (sausage-todd) [inline backend/src/database/migrations/V1719390926__orgAvgContributorEngagement.sql:1]: Let's do it this way:
1. Have the field non-nullable like every other aggregated field in this table
2. So, no arbitrary default values, which if they come up at some point in the UI, would look confusing
3. So, drop the default value here
4. In the end of the migration, after populating values, make the field non-nullable

**Review by sausage-todd — COMMENTED** [review]
Looks good apart from the nullable field

File: backend/src/database/migrations/V1719390926__orgAvgContributorEngagement.sql (hunk 2 above)
Comment (yeganathan18) [reply backend/src/database/migrations/V1719390926__orgAvgContributorEngagement.sql:1]: Quick question: If we make it non-nullable, shouldn’t we handle it in the code if the aggregation is null? 

For example, if the `avgContributorEngagement` is null for an organization and segment, can we use COALESCE and fallback to 0?

Comment (sausage-todd) [reply backend/src/database/migrations/V1719390926__orgAvgContributorEngagement.sql:1]: If there are activities, aggregation can’t be null. If there are no activities, the org/segment will be excluded from the query.

**Pull request merged**